dynamic_rest/renderers.py

```diff
@@ -0,0 +1,93 @@
+from collections import defaultdict
+from rest_framework import renderers
+
+
+class DynamicJSONRenderer(renderers.JSONRenderer):
+
+  def render(self, data, accepted_media_type=None, renderer_context=None):
+    # if data is a string, just render it out using the default method
+    # this usually indicates an error message
+    if isinstance(data, basestring):
+      return super(DynamicJSONRenderer, self).render(data, accepted_media_type, renderer_context)
+
```

Comment:
In the case that `data` is a dict or a list, we will waste a bit of time traversing through the structure looking for `_pk`s and `_model`s but given that an error dict is shallow, this shouldn't be much of a problem; in the end, we will call the superclass anyway to render the same result. Essentially, `_sideload` already does this check.